tests/func/test_update.py

```diff
@@ -71,3 +73,33 @@ def test_update_import_url(tmp_dir, dvc, tmp_path_factory):
 
     assert dst.is_file()
     assert dst.read_text() == "updated file content"
+
+
+def test_update_rev(tmp_dir, dvc, erepo_dir, monkeypatch):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.scm.checkout("new_branch", create_new=True)
```

Comment:
Another useful helper to change to a branch temporarely:
```suggestion
        with erepo_dir.branch("new_branch", new=True):
            # ...
```